native/game_backend/src/space_hash_grid.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{map::Position, player::Player, projectile::{Projectile, self}};
+use rustler::NifMap;
+use std::collections::HashMap;
+
+#[derive(Clone, Debug, rustler::NifMap)]
```

Comment:
For this simple type, `Copy` may be quite handy.
```suggestion
#[derive(Clone, Copy, Debug, rustler::NifMap)]
```